[stable/datadog] Security Agent settings for compliance and runtime security

xornivore opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->
#### Is this a new chart
> NOTE: We're experiencing a high volume of PRs to this repo and reviews will be delayed. Please host your own chart repository and submit your repository to the Helm Hub instead of this repo to make them discoverable to the community. [Here](https://github.com/helm/hub/blob/master/Repositories.md) is how to submit new chart repositories to the Helm Hub.

#### What this PR does / why we need it:

Changes to enable running security agent container for compliance monitoring and runtime security.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

Commit:
[stable/datadog] Helm chart changes to introduce security agent and compliance checks

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Guard config map mount in DCA by .Values.datadog.securityAgent.compliance.enabled

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Fix env var and add psp to RBAC

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

vboulineau commented on stable/datadog/README.md:
```diff
@@ -328,6 +328,10 @@ helm install --name <RELEASE_NAME> \
 | `datadog.systemProbe.enableConntrack`                        | If true, system-probe connects to the netlink/conntrack subsystem to add NAT information to connection data. Ref: http://conntrack-tools.netfilter.org/                             | `true`                                          |
 | `datadog.systemProbe.bpfDebug`                               | If true, system-probe writes debug logs to /sys/kernel/debug/tracing/trace_pipe                                                                                                     | `false`                                         |
 | `datadog.systemProbe.apparmor`                               | Apparmor profile for system-probe                                                                                                                                                   | `unconfined`                                    |
+| `datadog.securityAgent.enabled`                              | Enable security agent                                                                                                                                                               | `false`                                         |
```
Should not exist, can be deduced from `.compliance.enabled || .security.enabled` I guess

xornivore replied:
This is following the approach similar to `processAgent` - where the container is enabled and then features are configured for it.

vboulineau commented on stable/datadog/templates/agent-rbac.yaml:
```diff
@@ -18,6 +18,7 @@ rules:
   - pods
   - nodes
   - componentstatuses
+  - podsecuritypolicies
```
`podsecuritypolicies` belongs to the `policy` group

xornivore replied:
Ah good point, thank you!

Commit:
Fixes for security agent compliance checks

- Remove `securityAgent.enabled` to prevent possible misconfiguration.
- Use `securityAgent.compliance.enabled` to turn on compliance and security agent container.
- Add Cluster Agent configuration env vars to the security agent container.
- Fix PSP RBAC in cluster role.

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Fix mis-merge

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Add CI values for security agent for compliance

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

ahmed-mez commented on stable/datadog/templates/container-security-agent.yaml:
```diff
@@ -0,0 +1,72 @@
+    - name: group
+      mountPath: /etc/group
+    - name: hostroot
+      mountPath: /host/root
```
Let's add `readOnly: true`

Commit:
Mount system paths readonly

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Add support for runtime security configuration

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Drop CHANGELOG changes - should be automatically generated from git history

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Automatically enable system probe if securityAgent.runtime is enabled, set runtime policies dir volume to an emptyDir as a safe default

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

Commit:
Remove securityAgent.runtime.socket from user facing configuration in values

Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>
